Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -399,9 +400,11 @@ ClipmapSample SampleClipmap(float2 worldPosition)
             GetMacroClipmapCenterInWorldSpace(macroClipmapLevel.m_closestLevel)
         );
 
-        data.m_macroColor = ColorBilinearSampling(
+        float4 macroColor = ColorBilinearSampling(
             TerrainSrg::m_macroColorClipmaps, macroClipmapUvs, macroClipmapLevel.m_closestLevel,
-            TerrainSrg::m_clipmapData.m_enableMacroClipmapOverlay).rgb;
+            TerrainSrg::m_clipmapData.m_enableMacroClipmapOverlay);
+        data.m_macroColor = macroColor.rgb;
+        data.m_hasMacroNormal = macroColor.a == 1.0;
         data.m_macroNormal = NormalBilinearSampling(TerrainSrg::m_macroNormalClipmaps, macroClipmapUvs, macroClipmapLevel.m_closestLevel);
```

Comment:
nit: might be worth not calling `NormalBilinearSampling()` when hasMacroNormal is false. I'd expect all threads take the same path for this most of the time.

Reply:
Same for below